From stable-bounces@linux.kernel.org Thu Feb  8 14:22:30 2007
From: Soeren Sonnenburg <kernel@nn7.de>
Date: Thu, 08 Feb 2007 14:20:38 -0800
Subject: enable mouse button 2+3 emulation for x86 macs
To: torvalds@linux-foundation.org
Cc: kernel@nn7.de, benh@kernel.crashing.org, paulus@samba.org, dtor@mail.ru, akpm@linux-foundation.org, stable@kernel.org
Message-ID: <200702082220.l18MKd82028502@shell0.pdx.osdl.net>


From: Soeren Sonnenburg <kernel@nn7.de>

As macbook/macbook pro's also have to live with a single mouse button the
following patch just enables the Macintosh device drivers menu in Kconfig +
adds the macintosh dir to the obj-* to make macbook* users happy (who use
exactly that since months....

Signed-off-by: Soeren Sonnenburg <kernel@nn7.de>
Cc: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Cc: Paul Mackerras <paulus@samba.org>
Cc: Dmitry Torokhov <dtor@mail.ru>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/Makefile          |    2 +-
 drivers/macintosh/Kconfig |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- linux-2.6.20.1.orig/drivers/Makefile
+++ linux-2.6.20.1/drivers/Makefile
@@ -30,7 +30,7 @@ obj-$(CONFIG_PARPORT)		+= parport/
 obj-y				+= base/ block/ misc/ mfd/ net/ media/
 obj-$(CONFIG_NUBUS)		+= nubus/
 obj-$(CONFIG_ATM)		+= atm/
-obj-$(CONFIG_PPC_PMAC)		+= macintosh/
+obj-y				+= macintosh/
 obj-$(CONFIG_IDE)		+= ide/
 obj-$(CONFIG_FC4)		+= fc4/
 obj-$(CONFIG_SCSI)		+= scsi/
--- linux-2.6.20.1.orig/drivers/macintosh/Kconfig
+++ linux-2.6.20.1/drivers/macintosh/Kconfig
@@ -1,6 +1,6 @@
 
 menu "Macintosh device drivers"
-	depends on PPC || MAC
+	depends on PPC || MAC || X86
 
 config ADB
 	bool "Apple Desktop Bus (ADB) support"
